native/src/main/java/io/ballerina/wso2/apim/catalog/utils/Utils.java

```diff
@@ -0,0 +1,145 @@
+import static io.ballerina.wso2.apim.catalog.utils.Constants.SLASH;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.UTF8;
+
+/**
+ * Utility functions used in the service catalog.
+ *
+ * @since 0.1.0
+ */
+public class Utils {
+    public static String createMd5Hash(String string) {
+        try {
+            byte[] bytesOfMessage = string.getBytes(UTF8);
+            MessageDigest md = MessageDigest.getInstance(MD5_ALGO_NAME);
+            return new String(md.digest(bytesOfMessage), StandardCharsets.UTF_8);
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String getSecurityType(BMap<BString, Object> httpAnnotation) {
+        if (httpAnnotation.containsKey(StringUtils.fromString(AUTH))) {
+            BArray authArray = httpAnnotation.getArrayValue(StringUtils.fromString(AUTH));
+            for (Object authObject: authArray.getValues()) {
+                BMap<BString, Object> auth = (BMap<BString, Object>) authObject;
+                if (auth.containsKey(StringUtils.fromString(OAUTH2_AUTH))
+                        || auth.containsKey(StringUtils.fromString(JWT_AUTH))) {
+                    return OAUTH2;
+                }
+            }
+            return BASIC;
+        }
+        return NONE;
+    }
+
+    public static String getDefinitionType() {
+        return OAS3;
+    }
+
+    public static String getHostname(BMap<BString, Object> annotation) {
+        if (annotation == null || !annotation.containsKey(StringUtils.fromString(HOST))) {
+            return LOCALHOST;
+        }
+        return StringUtils.getStringValue(annotation.get(StringUtils.fromString(HOST)));
+    }
+
+    public static OpenAPI getOpenApiDefinition(BMap<BString, Object> annotation) {
+
+        BArray openApiDef = (BArray) annotation.get(StringUtils.fromString(OPENAPI_DEFINITION));
+        byte[] openApiDefByteStream = openApiDef.getByteArray();
+        String string = new String(openApiDefByteStream, StandardCharsets.UTF_8);
+        ParseOptions parseOptions = new ParseOptions();
+        parseOptions.setResolve(true);
+        parseOptions.setResolveFully(true);
+        return new OpenAPIParser().readContents(string, null, parseOptions).getOpenAPI();
+    }
+
+    public static String generateBasePath(BArray attachPointsArray) {
+        if (attachPointsArray == null) {
+            return SLASH;
+        }
+
+        String[] attachPoints = attachPointsArray.getStringArray();
+        if (attachPoints.length == 0) {
+            return SLASH;
+        }
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(SLASH);
+        for (int i = 0; i < attachPoints.length; i++) {
+            sb.append(attachPoints[i]).append(SLASH);
+        }
+        return sb.substring(0, sb.length() - 1);
```

Comment:
Assuming you intention is to add `SLASH` as prefix and join each with same I think using the `StringJoiner` is better (Also you can use stream with join collector as well)
```suggestion
        StringJoing sj = new StringJoiner(SLASH, SLASH, "");
        for (int i = 0; i < attachPoints.length; i++) {
            sb.append(attachPoints[i]);
        }
        return sj.toString();
```